virtualenv_embedded/activate.ps1

```diff
@@ -51,10 +51,22 @@ if (!$env:VIRTUAL_ENV_DISABLE_PROMPT)
         ""
     }
     $function:_old_virtual_prompt = $function:prompt
-    function global:prompt
+    if ("__VIRTUAL_PROMPT__" -ne "")
     {
-        # Add a prefix to the current prompt, but don't discard it.
-        write-host "($( split-path $env:VIRTUAL_ENV -leaf )) " -nonewline
-        & $function:_old_virtual_prompt
+        function global:prompt
+        {
+            # Add the custom prefix to the existing prompt
+            write-host "__VIRTUAL_PROMPT__" -nonewline
+            & $function:_old_virtual_prompt
+        }
+    }
+    else
+    {
+        function global:prompt
+        {
+            # Add a prefix to the current prompt, but don't discard it.
+            write-host "($( split-path $env:VIRTUAL_ENV -leaf )) " -nonewline

```

Comment:
maybe we just define a variable within the if to contain the text, and have a single prompt function, not? I assume there's some way to bind in the string as an argument to the function 🤔

Reply:
Mm, make it a default value of an argument or something?

This could maybe be done for most of the shells, actually -- e.g., a new alias in the `activate.sh` that gets configured to return the appropriate prompt string, and then the `$PS1` makes use of it somehow?

That seems out of scope for this PR, though.

Reply:
Thinking about it, I think it would make more sense to do this by changing the logic in virtualenv.py itself: basically, to change all of the activate scripts to do [what __VIRTUAL_WINPROMPT__ does](https://github.com/pypa/virtualenv/blob/94c6bcb4a3f8d7ff63e2215b18d76514d147a762/virtualenv.py#L1696). Currently I think only Win cmd uses this `WINPROMPT` template substitution; the others use the [`__VIRTUAL_PROMPT__`](https://github.com/pypa/virtualenv/blob/94c6bcb4a3f8d7ff63e2215b18d76514d147a762/virtualenv.py#L1695) template substitution.

I mean, the virtualenv name/folder/etc. are hardcoded and set at the time of environment creation, so there should be little lost by also hardcoding the prompt substitution into the activate scripts.

Reply:
This might munge the `--relocatable` capability, though..?

Reply:
I strongly prefer this be considered as out of scope for this PR. My goal with the PR is only to change and put under test the cosmetics of the prompts, and this seems like it has the potential to dig into subtle behavioral aspects of the code.

Reply:
ok